### PR TITLE
Add State class, add specs

### PR DESCRIPTION
**Pull request by chriswoodrich**

### CODE REVIEW

File: lib/lita/robot.rb [hunk 1]
```diff
@@ -63,8 +69,9 @@ class Robot
     # @param registry [Registry] The registry for the robot's configuration and plugins.
     def initialize(registry = Lita)
       @registry = registry
-      @name = config.robot.name
+      @name         = config.robot.name
```
Comment (jimmycuadra) [inline lib/lita/robot.rb]: No spacing change in the constructor, por favor.

File: lib/lita/robot.rb [hunk 2]
```diff
@@ -44,6 +45,11 @@ class Robot
     # @since 4.0.0
     attr_reader :registry
 
+    # The {State} for the currently running robot.
+    # @return [State] The state.
+    # @since 5.0.0
+    attr_accessor :state
```
Comment (jimmycuadra) [inline lib/lita/robot.rb]: Remove this, no need to expose it directly.

**Comment by tachyons** [comment]
Why this issue is closed ?

**Comment by jimmycuadra** [comment]
@tachyons It was implemented in https://github.com/litaio/lita/commit/8a1cfd94e174e7542b2d1210a3e8c01724aac637.

**Pull request closed without merging**